crates/primitives/src/transaction/sidecar.rs

```diff
@@ -503,3 +503,154 @@ fn generate_blob_sidecar(blobs: Vec<Blob>) -> BlobTransactionSidecar {
 
     BlobTransactionSidecar { blobs, commitments, proofs }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        hex,
+        kzg::{Blob, Bytes48},
+        transaction::sidecar::generate_blob_sidecar,
+        BlobTransactionSidecar,
+    };
+    use std::{fs, path::PathBuf};
+
+    #[test]
+    fn test_blob_transaction_sidecar_generation() {
+        // Read the contents of the JSON file into a string.
+        let json_content = fs::read_to_string(
+            PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/transaction/blob_data/blob1.json"),
+        )
+        .expect("Failed to read the blob data file");
+
+        // Parse the JSON contents into a serde_json::Value
+        let json_value: serde_json::Value =
+            serde_json::from_str(&json_content).expect("Failed to deserialize JSON");
+
+        // Extract blob data from JSON and convert it to Blob
+        let blobs: Vec<Blob> = vec![Blob::from_hex(
+            json_value.get("data").unwrap().as_str().expect("Data is not a valid string"),
```

Comment:
where is the json format from? was it hand-rolled or is it from an API somewhere? If hand-rolled, maybe this could instead just be hand rolled into the `BlobTransactionSidecar` json format. If it's from an API, I'd consider adding a test helper type that derives `Serialize, Deserialize` instead of using `json.Value`

Reply:
there's an api spec:

https://api.blobscan.com/

I can create a good first issue on the block explorers crate for adding a blob client

Reply:
> there's an api spec:
> 
> https://api.blobscan.com/
> 
> I can create a good first issue on the block explorers crate for adding a blob client

ah yeah, that would be great

Reply:
@Rjected Indeed it is done by hand here, because I do not need all the data (proof for example as long as I generate the `BlobTransactionSidecar` via `Blob` uniquely).

For the blob for which I am testing the rlp, I also stored in the JSON the rlp that I calculated myself same aside to make the assertion, so this rlp (which is too big to be in a `.rs` file does not fit in the `BlobTransactionSidecar` spec). 

For this PR, do we stay like this knowing that the client blob will arrive with the next issue or do you want me to make changes?

Reply:
> For this PR, do we stay like this knowing that the client blob will arrive with the next issue or do you want me to make changes?

Yeah, I think this is ok for now